lib/pbench/agent/tool_meister.py

```diff
@@ -200,210 +347,141 @@ def stop(self):
+    directed for transient tools.
     """
 
-    def __init__(self, name, tool_opts, logger=None, **kwargs):
-        super().__init__(name, tool_opts, logger=logger, **kwargs)
-        if self.tool_dir:
-            self.tool_dir = self.tool_dir / self.name.replace("-transient", "")
-        if "/usr/libexec/pcp/bin" not in os.environ["PATH"]:
-            os.environ["PATH"] += os.pathsep + "/usr/libexec/pcp/bin"
-        pmcd_path = find_executable("pmcd")
-        if pmcd_path:
-            self.pmcd_args = [
-                pmcd_path,
-                "--foreground",
-                "--socket=./pmcd.socket",
-                "--port=55677",
-                f"--config={self.pbench_install_dir}/templates/pmcd.conf",
-            ]
-        else:
-            self.pmcd_args = None
-        pmlogger_path = find_executable("pmlogger")
-        if pmlogger_path:
-            self.pmlogger_args = [
-                pmlogger_path,
-                "--log=-",
-                "--report",
-                "-t",
-                "3s",
-                "-c",
-                f"{self.pbench_install_dir}/templates/pmlogger.conf",
-                "--host=localhost:55677",
-                f"{self.tool_dir}/%Y%m%d.%H.%M",
-            ]
-        else:
-            self.pmlogger_args = None
+    def __init__(self, name, tool_opts, **kwargs):
+        super().__init__(name, tool_opts, **kwargs)
+        self.real_name = self.name.replace("-transient", "")
+        self.pmcd_args = None
         self.pmcd_process = None
+        self.pmlogger_args = None
         self.pmlogger_process = None
-        if self.tool_dir:
-            try:
-                self.tool_dir.mkdir()
-            except Exception as exc:
-                self.logger.error(
-                    "Failed to create tool directory '%s': '%s'", self.tool_dir, exc
-                )
+        if "/usr/libexec/pcp/bin" not in os.environ["PATH"]:
+            # FIXME - Shouldn't this be provided by the environment?
+            os.environ["PATH"] += ":/usr/libexec/pcp/bin"
+        self.pmcd_path = find_executable("pmcd")
+        self.pmlogger_path = find_executable("pmlogger")
 
     def install(self):
-        if self.pmcd_args is None:
+        if not self.pmcd_path:
             return (1, "pcp tool (pmcd) not found")
-        elif self.pmlogger_args is None:
+        if not self.pmlogger_path:
             return (1, "pcp tool (pmlogger) not found")
-        return (0, "pcp tool (pmcd and pmlogger properly installed")
+        return (0, "pcp tool (pmcd and pmlogger) properly installed")
 
     def start(self):
-        assert self.tool_dir is not None, "Logic bomb!  no tool directory provided!"
-        if self.pmcd_process or self.pmlogger_process:
+        if not self.pmcd_path:
+            raise RuntimeError("Path to pmcd not provided")
+        if not self.pmlogger_path:
+            raise RuntimeError("Path to pmlogger not provided")
```

Comment:
Are these checks necessary?  (They look redundant with`install()`....)

Reply:
Yes, all this is fixed in PR #2697.